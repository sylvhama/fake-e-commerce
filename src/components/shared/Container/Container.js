import styled from 'styled-components';
import PropTypes from 'prop-types';

const Container = styled.div`
  margin: 0 auto;
  padding: ${props =>
    `${props.noGutter ? 0 : props.theme.spacing.reg} ${
      props.noGutter ? 0 : props.theme.spacing.sm
    } 0`};
  max-width: ${props =>
    props.maxWidth ? props.theme.size.container[props.maxWidth] : null};
`;

Container.propTypes = {
  maxWidth: PropTypes.oneOf(['reg', 'lg']),
  noGutter: PropTypes.bool
};

export default Container;
