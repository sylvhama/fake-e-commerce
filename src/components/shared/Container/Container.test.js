import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../tests/helpers';

import Container from './Container';

test('Container renders correctly', () => {
  const rendered = renderWithTheme(<Container>Container</Container>).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('padding', '16px 8px 0');
  expect(rendered).toHaveStyleRule('max-width', undefined);
});

test('Container renders correctly', () => {
  const rendered = renderWithTheme(
    <Container noGutter maxWidth="reg">
      Container
    </Container>
  ).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('padding', '0 0 0');
  expect(rendered).toHaveStyleRule('max-width', '940px');
});
