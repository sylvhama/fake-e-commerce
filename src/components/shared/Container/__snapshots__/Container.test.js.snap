// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
.c0 {
  margin: 0 auto;
  padding: 16px 8px 0;
}

<div
  className="c0"
>
  Container
</div>
`;

exports[`Container renders correctly 2`] = `
.c0 {
  margin: 0 auto;
  padding: 0 0 0;
  max-width: 940px;
}

<div
  className="c0"
>
  Container
</div>
`;
